App/Sagas/TextileSagas.js

```diff
@@ -141,8 +143,19 @@ export function * stopNode () {
 
 export function * getPhotoHashes ({thread}) {
   try {
-    const photoData = yield call(IPFS.getPhotos, null, -1, thread)
-    yield put(IpfsNodeActions.getPhotoHashesSuccess(thread, photoData))
+    const hashes = yield call(IPFS.getPhotos, null, -1, thread)
+    let data = []
+    for (const hash of hashes) {
+      try {
+        const meta = yield call(IPFS.getHashData, hash, '/caption')
+        const caption = JSON.parse(Buffer.from(meta, 'base64').toString('ascii'))
+        data.push({hash, caption})
+      } catch (err) {
+        // gracefully return an empty caption for now
+        data.push({ hash, caption: '' })
```

Comment:
Seems like setting `null` here would be better so we can do more simple checks to decide whether to render the comment component:
`if (caption)` instead of `if (caption && caption.length > )`